koch.nim

```diff
@@ -12,23 +12,22 @@ Options:
   --nim:path                  use the specified NimSkull compiler
 
 Commands:
-  all [args]                  builds all programs
-  single <name> [args]        builds the single program with the given name
-  generate [dir]              generates the various language-related modules
+  build <target> [<arg> ...]  builds a program with the given arguments. 'all'
+                              builds all programs
+  generate <dir>              generates the various language-related modules
   build-defs                  verifies the language definitions and generates
                               the textual representation for them
 """
-  Programs: seq[(string, string, bool, bool)] = @[
-    ("tester", "tools/tester.nim", true, true),
-    ("passtool", "tools/passtool/passtool.nim", true, true),
-    ("queryshell", "phy/queryshell.nim", true, true),
-    ("repl", "phy/repl.nim", false, true),
-    ("phy", "phy/phy.nim", false, true),
-    ("skully", "skully/skully.nim", true, false)
-    # ^^ excluded from 'all' because the program takes too long to compile
-  ]
-    ## program name, module path, whether the program doesn't depend on
-    ## generated modules, and whether the program is built with 'all'
+  Programs = {
+    "tester" : ("tools/tester.nim", true),
+    "passtool" : ("tools/passtool/passtool.nim", true),
+    "queryshell" : ("phy/queryshell.nim", true),
+    "repl" : ("phy/repl.nim", false),
+    "phy" : ("phy/phy.nim", false),
+    "skully" : ("skully/skully.nim", true),
+  }.toOrderedTable
```

Comment:
```suggestion
  type
    Program = enum
      testerProg = "tester"
      passtoolProg = "passtool"
      queryshellProg = "queryshell"
      replProg = "repl"
      phyProg = "phy"
      skullProg = "skully"

  Programs: = [
    testerProg : ("tools/tester.nim", true),
    passtoolProg : ("tools/passtool/passtool.nim", true),
    queryshellProg : ("phy/queryshell.nim", true),
    replProg : ("phy/repl.nim", false),
    phyProg : ("phy/phy.nim", false),
    skullyProg : ("skully/skully.nim", true),
  ]
```

You might have already considered this, in which case feel free to disregard, I just lean towards more static things.

Reply:
I don't think it makes sense here, as `koch` doesn't need to refer to individual programs itself (except for `skully`, but that's an exception) and all access of the table uses a user-supplied string value anyway.

If `koch` grows more special cases for individual programs (though I think it shouldn't), then it could be a good idea to use an enum, but right now, I don't see the benefit of using one.